packages/ink-contracts/src/ink-client.ts

```diff
@@ -101,19 +161,26 @@ export const getInkClient = <
   }
 }
 
+const getMetadata = (spec: ConstructorSpec | MessageSpec) => ({
```

Comment:
Could possibly makes this generic? Useful since we already know the metadata upfront.

Reply:
What do you mean? This is just normalising the interface between constructors and messages, adding the assumption that constructors are mutates=true

Reply:
I mean something like this (doesn't really make sense, just to illustrate what I mean):

```ts
<TDefault extends boolean, TPayable extends boolean, TMutates extends boolean>(/* ... */) => { /* ... */ }
```

So that the metadata types will be a `const` instead of i.e. `boolean`.

Reply:
This function is just getting the run-time value, these types would have to be done to the interface:

<img width="699" alt="image" src="https://github.com/user-attachments/assets/d05ff3cd-2347-4aee-a5d5-19f275f70ca2" />

And... I think it's a bit overkill for the ink-client. If later we find it's useful enough we can always add them later.

Reply:
Yeah, I agree on the potentially "overkill" part. I also haven't thought of a particularly useful case for them myself. I was thinking if it cost nothing to includes those types info, then might as well :p